packages/mermaid/src/diagrams/pie/pieRenderer.ts

```diff
@@ -0,0 +1,179 @@
+import d3, { scaleOrdinal, pie as d3pie, arc } from 'd3';
+
+import { log } from '../../logger.js';
+import { configureSvgSize } from '../../setupGraphViewbox.js';
+import { getConfig } from '../../config.js';
+import { cleanAndMerge, parseFontSize } from '../../utils.js';
+import type { DrawDefinition, Group, SVG } from '../../diagram-api/types.js';
+import type { D3Sections, PieDB, Sections } from './pieTypes.js';
+import type { MermaidConfig, PieDiagramConfig } from '../../config.type.js';
+import { selectSvgElement } from '../../rendering-util/selectSvgElement.js';
+
+const createPieArcs = (sections: Sections): d3.PieArcDatum<D3Sections>[] => {
+  // Compute the position of each group on the pie:
+  const pieData: D3Sections[] = Object.entries(sections).map(
+    (element: [string, number]): D3Sections => {
+      return {
+        label: element[0],
+        value: element[1],
+      };
+    }
+  );
+  const pie: d3.Pie<unknown, D3Sections> = d3pie<D3Sections>().value(
+    (d3Section: D3Sections): number => d3Section.value
+  );
+  return pie(pieData);
+};
+
+/**
+ * Draws a Pie Chart with the data given in text.
+ *
+ * @param text - pie chart code
+ * @param id - diagram id
+ * @param _version - MermaidJS version from package.json.
+ * @param diagObj - A standard diagram containing the DB and the text and type etc of the diagram.
+ */
+export const draw: DrawDefinition = (text, id, _version, diagObj) => {
+  log.debug('rendering pie chart\n' + text);
+
+  const db = diagObj.db as PieDB;
+  const globalConfig: MermaidConfig = getConfig();
+  const pieConfig: Required<PieDiagramConfig> = cleanAndMerge(db.getConfig(), globalConfig.pie);
```

Comment:
Should this be the order of precedence?

I think the order of merging should be. 

1. Default config
2. Global config
3. Directive config

Reply:
Not sure I'm following, do we need to merge directive configs here? The first two are as you stated.

Reply:
| Config    | Eg1 | Eg2 | Eg3 | Eg4 |
| --------- | --- | --- | --- | --- |
| default   | 1   | 1   | 1   | 1   |
| global    | -   | -   | 2   | 2   |
| directive | -   | 2   | -   | 3   |
| final  value   | 1   | 2   | 2   | 3   |

An example of how the final config value of a field should be, if it's defined in multiple places.

Reply:
Sorry I didn't make it clear, my argument was referring to that directives don't have any common stuff with default and global values, but I forgot that the pie directives also have `textPosition`.

Reply:
How do we get directives so we could assign it?

Reply:
What about `title`? Should directive/yaml override the one in diagram? If so, then current approach isn't doing so:

```
---
title: awesome
---
pie title cool
"section": 1
```


```mermaid
---
title: awesome
---
pie title cool
"section": 1
```

Reply:
> Personally, I feel like the one defined in the JISON should override the one defined in the YAML

So am I, I think the metadata defined in actual diagram should override the one YAML.